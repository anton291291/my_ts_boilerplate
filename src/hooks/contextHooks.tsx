import React, { useState, createContext } from 'react';
export const GridContext = createContext({} as GridContextType);

export declare namespace GlobalState {
    type CellItemType = { index: number; isClicked: boolean };

    type AxisType = {
        axis: {
            x: number;
            y: number;
        };
    };

    type CellType = {
        cells: Array<CellItemType>;
    };

    type IntervalIDType = {
        intervalID: number;
    };

    type isPlayType = {
        isPlay: boolean;
        speed: number;
    };

    type FormType = {
        name: string;
    };

    type GenType = {
        gen: number;
    };
}

export type StateType = GlobalState.AxisType &
    GlobalState.CellType &
    GlobalState.IntervalIDType &
    GlobalState.GenType &
    GlobalState.isPlayType &
    GlobalState.FormType;

type GridContextType = {
    state: StateType;
    setState: React.Dispatch<React.SetStateAction<StateType>>;
};

export const GridProvider = ({ children }) => {
    const y = 15;
    const x = 30;

    const [state, setState] = useState({
        axis: { x: x, y: y },
        cells: Array.from({ length: y * x }, (item, index) => ({
            index: index,
            isClicked: false
        })),
        gen: 1,
        name: '',
        intervalID: null,
        isPlay: false,
        speed: 1
    });

    return (
        <GridContext.Provider value={{ state, setState }}>
            {children}
        </GridContext.Provider>
    );
};
