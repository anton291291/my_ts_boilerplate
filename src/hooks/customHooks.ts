const useLogin = (params) => {
const [state, setstate] = useState(initialState)
}