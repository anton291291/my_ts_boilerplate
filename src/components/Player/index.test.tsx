import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { GridProvider } from '../../hooks/contextHooks';
import { Player } from '.';
import { mountToJson } from 'enzyme-to-json';

describe('Player', () => {
    const output = mount(
        <GridProvider>
            <Player />
        </GridProvider>
    );

    test('should render correctly', () => {
        expect(mountToJson(output)).toMatchSnapshot();
    });

    test('Checking generation default state', () => {
        expect(output.find('GenField').text()).toEqual('Generation: 1');
    });

    test('Checking Player  Play/Pause btns', () => {
        output.find('Random').simulate('click');
        output.find('Play').simulate('click');
        expect(output.find('Pause').exists()).toBeTruthy;
        output.find('Pause').simulate('click');
        expect(output.find('Play').exists()).toBeTruthy;
    });
});
