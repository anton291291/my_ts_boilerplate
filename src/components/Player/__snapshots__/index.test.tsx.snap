// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pleer should render correctly 1`] = `
<GridProvider>
  <Pleer>
    <Styled(MuiBox)
      display="flex"
    >
      <div
        className="MuiBox-root MuiBox-root-2"
      >
        <GenField>
          <WithStyles(ForwardRef(Typography))
            className="GenField-sc-9snjga-5 jDxALd"
          >
            <ForwardRef(Typography)
              className="GenField-sc-9snjga-5 jDxALd"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root GenField-sc-9snjga-5 jDxALd MuiTypography-body1"
              >
                Generation: 
                1
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </GenField>
        <Play
          onClick={[Function]}
        >
          <ForwardRef
            className="Play-sc-9snjga-2 iXLphk"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(SvgIcon))
              className="Play-sc-9snjga-2 iXLphk"
              onClick={[Function]}
            >
              <ForwardRef(SvgIcon)
                className="Play-sc-9snjga-2 iXLphk"
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root Play-sc-9snjga-2 iXLphk"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8 5v14l11-7z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </Play>
        <IconContainer>
          <div
            className="IconContainer-sc-9snjga-4 TBJdr"
          >
            <Stop
              onClick={[Function]}
            >
              <ForwardRef
                className="Stop-sc-9snjga-0 bggHgt"
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(SvgIcon))
                  className="Stop-sc-9snjga-0 bggHgt"
                  onClick={[Function]}
                >
                  <ForwardRef(SvgIcon)
                    className="Stop-sc-9snjga-0 bggHgt"
                    classes={
                      Object {
                        "colorAction": "MuiSvgIcon-colorAction",
                        "colorDisabled": "MuiSvgIcon-colorDisabled",
                        "colorError": "MuiSvgIcon-colorError",
                        "colorPrimary": "MuiSvgIcon-colorPrimary",
                        "colorSecondary": "MuiSvgIcon-colorSecondary",
                        "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                        "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                        "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                        "root": "MuiSvgIcon-root",
                      }
                    }
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root Stop-sc-9snjga-0 bggHgt"
                      focusable="false"
                      onClick={[Function]}
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 6h12v12H6z"
                      />
                    </svg>
                  </ForwardRef(SvgIcon)>
                </WithStyles(ForwardRef(SvgIcon))>
              </ForwardRef>
            </Stop>
          </div>
        </IconContainer>
        <Random
          onClick={[Function]}
        >
          <ForwardRef
            className="Random-sc-9snjga-1 kNTVjD"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(SvgIcon))
              className="Random-sc-9snjga-1 kNTVjD"
              onClick={[Function]}
            >
              <ForwardRef(SvgIcon)
                className="Random-sc-9snjga-1 kNTVjD"
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root Random-sc-9snjga-1 kNTVjD"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </Random>
        <NameField>
          <WithStyles(ForwardRef(Typography))
            className="NameField-sc-9snjga-6 gkgXIB"
          >
            <ForwardRef(Typography)
              className="NameField-sc-9snjga-6 gkgXIB"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root NameField-sc-9snjga-6 gkgXIB MuiTypography-body1"
              />
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </NameField>
      </div>
    </Styled(MuiBox)>
  </Pleer>
</GridProvider>
`;
